ddtrace/llmobs/_evaluators/ragas/faithfulness.py

```diff
@@ -13,24 +65,236 @@ class RagasFaithfulnessEvaluator:
 
-    def run_and_submit_evaluation(self, span: dict) -> None:
-        if not span:
+        self.split_answer_into_sentences = get_segmenter(
+            language=self.ragas_faithfulness_instance.nli_statements_message.language, clean=False
+        )
+
+    def run_and_submit_evaluation(self, span_event: dict):
+        if not span_event:
             return
-        score_result = self.evaluate(span)
+        score_result = self.evaluate(span_event)
         if score_result:
             self.llmobs_service.submit_evaluation(
-                span_context=span,
+                span_context={"trace_id": span_event.get("trace_id"), "span_id": span_event.get("span_id")},
                 label=RagasFaithfulnessEvaluator.LABEL,
                 metric_type=RagasFaithfulnessEvaluator.METRIC_TYPE,
                 value=score_result,
-                timestamp_ms=math.floor(time.time() * 1000),
             )
 
-    def evaluate(self, span: dict) -> Optional[float]:
-        """placeholder function"""
-        span_id, trace_id = span.get("span_id"), span.get("trace_id")
-        if not span_id or not trace_id:
+    def evaluate(self, span_event: dict) -> Optional[float]:
+        if not self.enabled:
             return None
-        return 1.0
+
+        """Get the latest faithfulness instance from Ragas"""
+        self.ragas_faithfulness_instance = _get_faithfulness_instance()
+
+        """Initialize defaults for variables we want to annotate on the LLM Observability trace of the
+        ragas faithfulness evaluations."""
+        score, question, answer, context, statements, faithfulness_list = math.nan, None, None, None, None, None
+
+        with self.llmobs_service.workflow("ragas.faithfulness", ml_app=_get_ml_app_for_ragas_trace(span_event)):
+            try:
+                """Get the necessary faithfulness inputs from the span event"""
+                faithfulness_inputs = self._extract_faithfulness_inputs(span_event)
+                if faithfulness_inputs is None:
+                    logger.debug(
+                        "Failed to extract question and context from span sampled for ragas_faithfulness evaluation"
+                    )
+                    return None
+
+                question = faithfulness_inputs["question"]
+                answer = faithfulness_inputs["answer"]
+                context = faithfulness_inputs["context"]
+
+                statements = self._create_statements(question, answer)
+                if statements is None:
+                    return None
+
+                faithfulness_list = self._create_verdicts(context, statements)
+                if faithfulness_list is None:
+                    return None
+
+                score = self._compute_score(faithfulness_list)
+                if math.isnan(score):
+                    return None
+
+                return score
+            finally:
+                self.llmobs_service.annotate(
+                    input_data=span_event,
```

Comment:
Just to be safe, can we pass the workflow span explicitly here to make sure we're annotating the correct one?